src/Command/PqCommands/ContentHubPqEntityStructure.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+namespace Acquia\Console\ContentHub\Command\PqCommands;
+
+use Acquia\Console\ContentHub\Command\Helpers\DrupalServiceFactory;
+use Symfony\Component\Console\Input\InputInterface;
+
+/**
+ * Runs a check against possibly unsupported entity types.
+ */
+class ContentHubPqEntityStructure extends ContentHubPqCommandBase {
+
+  /**
+   * Paragraph field type.
+   */
+  public const PARAGRAPH_FIELD = 'paragraph';
+
+  /**
+   * Entity reference field type.
+   */
+  public const ENTITY_REF = 'entity_reference';
+
+  /**
+   * Entity reference revision field type.
+   */
+  public const ENTITY_REF_REV = 'entity_reference_revisions';
+
+  /**
+   * Formatted text field types.
+   */
+  public const FORMATTED_TEXT_FIELDS = [
+    'text_with_summary',
+    'text',
+    'text_long',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $defaultName = 'ach:pq:entity-structure';
+
+  /**
+   * The Drupal service factory service.
+   *
+   * @var \Acquia\Console\ContentHub\Command\Helpers\DrupalServiceFactory
+   */
+  protected $serviceFactory;
+
+  /**
+   * Constructs a new ContentHubPqEntityTypes object.
+   *
+   * @param \Acquia\Console\ContentHub\Command\Helpers\DrupalServiceFactory $drupalServiceFactory
+   *   The Drupal service factory service.
+   * @param string|null $name
+   *   The name of this command.
+   */
+  public function __construct(DrupalServiceFactory $drupalServiceFactory, string $name = NULL) {
+    parent::__construct($name);
+    $this->serviceFactory = $drupalServiceFactory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function configure() {
+    parent::configure();
+    $this->setDescription('Analyses content type structure and sorts bundles based on their complexity.');
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @throws \Exception
+   */
+  protected function runCommand(InputInterface $input, PqCommandResult $result): int {
+    $bundles = $this->getNodeBundles();
+    $fieldData = $this->getFieldTypes(array_keys($bundles));
+    $kriName = 'Risky Content Types';
+    $riskyBundles = $this->analyseFieldData($bundles, $fieldData);
+
+    $formatted = [];
+    foreach ($fieldData as $bundleId => $bundleData) {
+      $formatted[$bundleData['complex_fields']['count'] ?? 0] = sprintf('%s: %s ', $bundles[$bundleId], $bundleData['complex_fields']['count'] ?? 0);
+    }
+    ksort($formatted);
+
+    $formatted = array_merge(['Content Type: Number of Complex fields'], $formatted);
+
+    $kriMessage = PqCommandResultViolations::$riskyBundles;
+    $result->setIndicator(
+      $kriName,
+      implode("\n", $formatted),
+      $kriMessage
+    );
+
+    if (!empty($riskyBundles)) {
+      $result->setIndicator(
+        'Content types with paragraphs',
+        implode("\n", $riskyBundles),
+        PqCommandResultViolations::$paragraphBundles,
+        TRUE
+      );
+    }
+    return 0;
```

Comment:
We can create separate KRI for every entity they want to check and we can default back to node. E.g.:
```
Content Type: Node  {list of bundles and number of risky fields}   {note}   {risky or not}
```